ambry-replication/src/main/java/com.github.ambry.replication/CloudToStoreReplicationManager.java

```diff
@@ -215,6 +219,42 @@ private void addCloudReplica(String partitionName) throws ReplicationException {
     }
   }
 
+  @Override
+  protected int reloadReplicationTokenIfExists(ReplicaId localReplicaId, List<RemoteReplicaInfo> remoteReplicaInfos)
+      throws ReplicationException {
+    int tokenReloadFailureCount = 0;
+    List<RemoteReplicaInfo.ReplicaTokenInfo> tokenInfos = persistor.retrieve(localReplicaId.getMountPath());
+    if (tokenInfos.size() != 0) {
+      for (RemoteReplicaInfo remoteReplicaInfo : remoteReplicaInfos) {
+        boolean tokenReloaded = false;
+        for (RemoteReplicaInfo.ReplicaTokenInfo tokenInfo : tokenInfos) {
+          // Note that in case of cloudReplicaTokens, the actual remote data node might not match as the node is chosen at
+          // random during initialization. So its enough to just match the partitionId in the token so that replication
+          // can start from cloud from where it left off.
+          if (tokenInfo.getReplicaInfo()
```

Comment:
I don't see the need to override the whole method which is mostly the same.  Can't you override isTokenForRemoteReplicaInfo() instead?